Userland/Libraries/LibCompress/Gzip.cpp

```diff
@@ -63,26 +63,19 @@ size_t GzipDecompressor::read(Bytes bytes)
             current_member().m_checksum.update(slice.trim(nread));
             current_member().m_nread += nread;
 
-            if (current_member().m_stream.handle_any_error()) {
-                set_fatal_error();
-                break;
-            }
+            if (current_member().m_stream.handle_any_error())
+                return Error::from_string_literal("Underlying DeflateDecompressor indicated an error");
 
             if (nread < slice.size()) {
                 LittleEndian<u32> crc32, input_size;
-                m_input_stream >> crc32 >> input_size;
+                TRY(m_input_stream->read(crc32.bytes()));
+                TRY(m_input_stream->read(input_size.bytes()));
 
-                if (crc32 != current_member().m_checksum.digest()) {
-                    // FIXME: Somehow the checksum is incorrect?
+                if (crc32 != current_member().m_checksum.digest())
+                    return Error::from_string_literal("CRC32 does not match the checksum of the current member");
```

Comment:
not sure if this error message is helpful enough… (“member”, checksum vs. crc32)

Reply:
`Error`s currently don't allow for storing non-literal strings, so I can't actually replace the "member" with anything useful.

However, I _can_ replace the checksum part with something like "Stored member CRC32 does not match the calculated CRC32".

That said, as long as errors don't allow for formatted strings and/or source information, I believe that those error messages are (for the most part) only useful for giving the developer an idea on where to roughly look and being easily greppable.